src/SpaceChat.ts

```diff
@@ -0,0 +1,171 @@
+import axios from 'axios'
+import fs from 'fs'
+import path from 'path'
+import readline from 'readline'
+import winston from 'winston'
+import { WebSocket } from 'ws'
+import { Downloader } from './Downloader'
+import { SpaceChatKind } from './enums/SpaceChatKind.enum'
+import { logger as baseLogger } from './logger'
+
+export class SpaceChat {
+  private readonly WS_URL = 'wss://prod-chatman-ancillary-ap-northeast-1.pscp.tv/chatapi/v1/chatnow'
+  private readonly CHAT_API_URL = 'https://proxsee.pscp.tv/api/v2/accessChatPublic'
+
+  private logger: winston.Logger
+  private ws: WebSocket
+  private accessChatData: Record<string, any>
+
+  private tmpChatFile: string
+  private outChatFile: string
+
+  constructor(
+    public spaceId: string,
+    private liveStreamStatus: Record<string, any>,
+    private options: Record<string, any>,
+  ) {
+    this.logger = baseLogger.child({ label: `[SpaceChat@${spaceId}]` })
+    this.spaceId = spaceId
+    this.liveStreamStatus = liveStreamStatus
+    this.options = options
+
+    this.tmpChatFile = path.join(Downloader.getMediaDir(this.username), `${this.filename} Chat.jsonl`)
+    this.outChatFile = path.join(Downloader.getMediaDir(this.username), `${this.filename} Chat.txt`)
+
+    Downloader.createMediaDir(this.username)
+    this.logger.info(`Tmp chat file ${this.tmpChatFile}`)
+    this.logger.info(`Out chat file ${this.outChatFile}`)
+  }
+
+  private get chatToken(): string {
+    return this.liveStreamStatus.chatToken
+  }
+
+  private get username(): string {
+    return this.options?.username || ''
+  }
+
+  private get filename(): string {
+    return this.options?.filename
+  }
+
+  public async watch() {
+    if (!this.chatToken) {
+      this.logger.warn('Chat token not found!')
+      return
+    }
+    try {
+      await this.getAccessChatData()
+      this.initWebSocket()
+    } catch (error) {
+      this.logger.error(error.message)
+    }
+  }
+
+  public unwatch() {
+    if (this.ws.readyState === WebSocket.OPEN) {
+      this.ws.close(0, 'Complete!')
+    }
+  }
+
+  private async getAccessChatData() {
+    const { data } = await axios.post<any>(this.CHAT_API_URL, { chat_token: this.chatToken })
+    this.logger.debug('accessChat data', data)
+    this.accessChatData = data
+  }
+
+  private initWebSocket() {
+    this.ws = new WebSocket(this.WS_URL, {})
+    this.attachWsEvents()
+  }
+
+  private attachWsEvents() {
+    const { ws } = this
+    ws.on('error', (error) => {
+      this.logger.error(`[WS] Error: ${error.message}`)
```

Comment:
Can make another child logger in this scope I guess.

Reply:
For now nothing much here so prob dont need 🤔